src/Kernel/BlockClosure.class.st

```diff
@@ -754,28 +754,27 @@ BlockClosure >> valueSupplyingAnswers: aListOfPairs [
 	"evaluate the block using a list of questions / answers that might be called upon to
 	automatically respond to Object>>confirm: or FillInTheBlank requests"
 
-	^ self 
+	^ self
 		on: ProvideAnswerNotification
-		do: 
-			[:notify | | answer caption |
-			
-			caption := notify messageText withSeparatorsCompacted. "to remove new lines"
-			answer := aListOfPairs
-				detect: 
-					[:each | caption = each first or:
-						[(caption includesSubstring: each first caseSensitive: false) or:
-						[(each first match: caption) or:
-						[(String includesSelector: #matchesRegex:) and: 
-						[ [ caption matchesRegex: each first ] on: Error do: [:ignored | false ]]]]]]
-					ifNone: [nil].
-			answer
-				ifNotNil: [notify resume: answer second]
-				ifNil: 
-					[ | outerAnswer |
-					outerAnswer := ProvideAnswerNotification signal: notify messageText.
-					outerAnswer 
-						ifNil: [notify resume] 
-						ifNotNil: [notify resume: outerAnswer]]]
+		do: [ :notify | 
+			| caption |
+			caption := notify messageText withSeparatorsCompacted.	"to remove new lines"
+			aListOfPairs
+				detect: [ :each | 
+					caption = each first
+						or: [ (caption includesSubstring: each first caseSensitive: false)
+								or: [ (each first match: caption)
+										or: [ (String includesSelector: #matchesRegex:)
+												and: [ [ caption matchesRegex: each first ]
+														on: Error
+														do: [ :ignored | false ] ] ] ] ] ]
+				ifFound: [ :answer | notify resume: answer second ]
+				ifNone: [ | outerAnswer |
+					outerAnswer := ProvideAnswerNotification
```

Comment:
Here we can also improve further changing it into:

```Smalltalk
(ProvideAnswerNotification signal: notify messageText) ifNil: [ notify resume ] ifNotNil: [ :outerAnswer | notify resume: outerAnswer ]
```

But it can be in a future PR :)